wdl-grammar/src/experimental/lexer/v1.rs

```diff
@@ -0,0 +1,1339 @@
+    #[token("input")]
+    InputKeyword,
+    /// The `meta` keyword.
+    #[token("meta")]
+    MetaKeyword,
+    /// The `null` keyword.
+    #[token("null")]
+    NullKeyword,
+    /// The `object` keyword.
+    #[token("object")]
+    ObjectKeyword,
+    /// The `output` keyword.
+    #[token("output")]
+    OutputKeyword,
+    /// The `parameter_meta` keyword.
+    #[token("parameter_meta")]
+    ParameterMetaKeyword,
+    /// The `runtime` keyword.
+    #[token("runtime")]
+    RuntimeKeyword,
+    /// The `scatter` keyword.
+    #[token("scatter")]
+    ScatterKeyword,
+    /// The `struct` keyword.
+    #[token("struct")]
+    StructKeyword,
+    /// The `task` keyword.
+    #[token("task")]
+    TaskKeyword,
+    /// The `then` keyword.
+    #[token("then")]
+    ThenKeyword,
+    /// The `true` keyword.
+    #[token("true")]
+    TrueKeyword,
+    /// The `version` keyword.
+    #[token("version")]
+    VersionKeyword,
+    /// The `workflow` keyword.
+    #[token("workflow")]
+    WorkflowKeyword,
+
+    /// The reserved `Directory` type keyword.
+    #[token("Directory")]
+    ReservedDirectoryTypeKeyword,
+    /// The reserved `hints` keyword.
+    #[token("hints")]
+    ReservedHintsKeyword,
+    /// The reserved `requirements` keyword.
+    #[token("requirements")]
+    ReservedRequirementsKeyword,
+
+    /// The `{` symbol.
+    #[token("{")]
+    OpenBrace,
+    /// The `}` symbol.
+    #[token("}")]
+    CloseBrace,
+    /// The `[` symbol.
+    #[token("[")]
+    OpenBracket,
+    /// The `]` symbol.
+    #[token("]")]
+    CloseBracket,
+    /// The `=` symbol.
+    #[token("=")]
+    Assignment,
+    /// The `:` symbol.
+    #[token(":")]
+    Colon,
+    /// The `,` symbol.
+    #[token(",")]
+    Comma,
+    /// The `(` symbol.
+    #[token("(")]
+    OpenParen,
+    /// The `)` symbol.
+    #[token(")")]
+    CloseParen,
+    /// The `?` symbol.
+    #[token("?")]
+    QuestionMark,
+    /// The `!` symbol.
+    #[token("!")]
+    Exclamation,
+    /// The `+` symbol.
+    #[token("+")]
+    Plus,
+    /// The `-` symbol.
+    #[token("-")]
+    Minus,
+    /// The `||` symbol.
+    #[token("||")]
+    LogicalOr,
+    /// The `&&` symbol.
+    #[token("&&")]
+    LogicalAnd,
+    /// The `*` symbol.
+    #[token("*")]
+    Asterisk,
+    /// The `/` symbol.
+    #[token("/")]
+    Slash,
+    /// The `%` symbol.
+    #[token("%")]
+    Percent,
+    /// The `==` symbol.
+    #[token("==")]
+    Equal,
+    /// The `!=` symbol.
+    #[token("!=")]
+    NotEqual,
+    /// The `<=` symbol.
+    #[token("<=")]
+    LessEqual,
+    /// The `>=` symbol.
+    #[token(">=")]
+    GreaterEqual,
+    /// The `<` symbol.
+    #[token("<")]
+    Less,
+    /// The `>` symbol.
+    #[token(">")]
+    Greater,
+    /// The `.` symbol.
+    #[token(".")]
+    Dot,
+
+    // WARNING: this must always be the last variant.
+    /// The exclusive maximum token value.
+    MAX,
+}
+
+// There can only be 128 tokens in a TokenSet.
```

Comment:
FYI: `TokenSet` will be introduced in an upcoming PR; basically the parser can reference sets of tokens in a bitfield, which will be useful for error recovery and also "expected one of these tokens"-style diagnostics.